Mastodon/Protocol/Provider/DataSourceFacade+SearchHistory.swift

```diff
@@ -8,11 +8,12 @@
 import Foundation
 import CoreDataStack
 import MastodonCore
+import UIKit
 
 extension DataSourceFacade {
     
     static func responseToCreateSearchHistory(
-        provider: DataSourceProvider & AuthContextProvider,
+        provider: NeedsDependency & UIViewController & AuthContextProvider,
```

Comment:
We're using `NeedsDependency & UIViewController` quite often, maybe it makes sense to create a `typealias` for this?

Reply:
it's.... 🥁 🥁 🥁 

🎉 ViewControllerWithDependencies🎉